ui/perfherder/graphs/TestDataModal.jsx

```diff
@@ -269,14 +262,7 @@ export default class TestDataModal extends React.Component {
       const tags = this.getTagOptions(seriesData);
 
       this.setState({ ...updates, tags });
-
-      if (tag !== this.defaultTag) {
-        await this.updateTaggedTests();
-      }
-
-      if (filterText) {
-        await this.updateFilterText(filterText);
-      }
+      this.applyFilters(filterText);
       return;
```

Comment:
I *think* we could eliminate this return statement & reorganize the code as a if/else construct.
That way, it's more obvious there are 2 main execution branches here.